packages/libs/react-ui/src/components/Accordion/Accordion.tsx

```diff
@@ -1,40 +1,96 @@
+import React, { useCallback, useEffect } from 'react';
 
-export interface IAccordionProps {
-  sections: Omit<IAccordionSectionProps, 'isOpen' | 'onToggle'>[];
+export interface IAccordionRootProps {
+  children?: FunctionComponentElement<IAccordionSectionProps>[];
   linked?: boolean;
+  openSection?: number;
 }
 
-export const Accordion: FC<IAccordionProps> = ({ sections, linked = true }) => {
-  const [expandedSections, setExpandedSections] = useState<number[]>([]);
-  const handleOpen = (index: number): void => {
-    if (linked) setExpandedSections([index]);
-    else setExpandedSections([...expandedSections, index]);
-  };
-  const handleClose = (index: number): void => {
-    setExpandedSections(expandedSections.filter((item) => item !== index));
-  };
+export const AccordionRoot: FC<IAccordionRootProps> = ({
+  children,
+  linked = false,
+  openSection,
+}) => {
+  const { usingLinked, setUsingLinked, openSections, setOpenSections } =
+    useLinked(openSection);
 
-  const isOpen = (index: number): boolean => expandedSections.includes(index);
+  useEffect(() => {
+    setUsingLinked(linked);
+    if (linked && openSections.length > 1) {
+      const lastOpen = openSections.pop() || -1;
+      setOpenSections([lastOpen]);
+    }
+  }, [linked]);
 
-  const handleToggle = (index: number): void => {
-    if (isOpen(index)) handleClose(index);
-    else handleOpen(index);
-  };
+  const handleToggleSection = useCallback(
+    (
+      index: number,
+      { onOpen, onClose }: Pick<IAccordionSectionProps, 'onOpen' | 'onClose'>,
+    ): void => {
+      const isOpen = openSections.includes(index);
+      if (isOpen) {
+        setOpenSections(openSections.filter((i) => i !== index));
+        onClose?.();
+      } else {
+        setOpenSections(usingLinked ? [index] : [...openSections, index]);
+        onOpen?.();
+      }
+    },
+    [openSections, usingLinked],
+  );
 
   return (
-    <div data-testid="kda-accordion-wrapper">
-      {sections.map((section, index) => (
-        <AccordionSection
-          {...section}
-          isOpen={isOpen(index)}
-          onToggle={() => handleToggle(index)}
-          key={String(section.title)}
+    <div data-testid="kda-accordion-sections">
+      {React.Children.map(children, (section, sectionIndex) => (
+        <section
+          className={accordionSectionClass}
+          data-testid="kda-accordion-section"
         >
-          {section.children}
-        </AccordionSection>
+          <div
+            data-testid="kda-accordion-section-title"
+            onClick={() =>
+              handleToggleSection(sectionIndex, {
+                onOpen: section?.props.onOpen,
+                onClose: section?.props.onClose,
+              })
+            }
+            className={classNames(
+              accordionTitleClass,
+              accordionTitleVariants[
+                openSections.includes(sectionIndex) ? 'opened' : 'closed'
+              ],
+            )}
```

Comment:
Why is this logic/styling in `Accordion.Root` instead of `Accordion.Section`?

Reply:
I kept the logic in Root that decides what is rendered and how by iterating over the children. Open to other ideas though! My focus was really how do I refactor the Accordion that takes a `sections` prop, array of sections, to an Accordion that takes subcomponents to render its sections.

Reply:
As discussed in previous PRs, I think it makes sense to colocate any styles and logic with the subcomponent that they are actually associated with. This helps with maintainability because it makes the code clearer and more predictable. Just how on links we would want the link itself to actually apply styles, not the parent - similar to why VE doesn't allow you to apply styles to other elements

Reply:
What do you think?

Reply:
Yeah that's true. I'll have another look at it!